text/0031-expanded_remove.md

```diff
@@ -0,0 +1,236 @@
+
+Adding an `--amend` flag has the UX advantage of working language that resembles "undo" directly into the save command, which users will be familiar with before they ever need `delete`. As a git user I (b5) learned to use `--amend` before `rebase`, which I admit wasn't great, but allowed me to stick with git long enough to understand what rebase did.
+
+Users should come to think of `--amend` as a shortcut:
+
+```
+# two commands:
+$ qri remove me/dataset --revisions 1
+$ qri save me/dataset
+
+# same thing, one command:
+$ qri save me/dataset --amend 
+```
+
+When a user understands these two things to be equal, it'll help solidfy their mental model of versioning with Qri.
+
+
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The remove command unpins blocks AND alters the logbook. Removing a read-only dataset drops logbook data entirely. Removing from a writable dataset adds an operation to the dataset log marking the number of versions removed.
+
+All the previously stated local delete behaviors stay the same, with a few exceptions listed below.
+But in addition to those behaviors, we have remote behaviors as well:
+```
+$ qri remove me/dataset --remote registry
+```
+Unpins all blocks & removes the log on the registry because the registry has a read-only copy. Does nothing locally.
+
+Attempting to drop revisions from a read-only dataset replica is an error:
+```
+$ qri remove me/dataset --remote registry --revisions 1
+ERROR
+```
+
+This you can't manipulate the logbook of a dataset on a remote. Instead you would first edit locally & push:
+```
+$ qri remove me/dataset --revisions 1
+$ qri push me/dataset
+```
+
+remove works without any flags on read-only datasets:
+```
+$ qri remove not_me/dataset
+```
+unpins the blocks and then removes the log of a dataset that is not in your namespace. Because we have to remove the log entirely, we cannot restore a dataset that was removed but not in your namespace. Instead restore by re-pulling.
+
+using revisions with a read-only dataset will error
+```
+$ qri remove not_me/dataset --revisions 1
+ERROR
+```
+should error: you can't manipulate the logbook of a read-only dataset.
+
+If you alter your logbook locally using qri remove, you can update the remote log by using qri push.
+
+
+### Use the logbook
+Delete actions should examine the history of a logbook to present warnings to a user that help them "clean up" when deleteing stuff. As an example: `$ qri remove --all` on a dataset should look for `push` operations and warn user they should `drop` from a list of remotes.
+
+### rename the `--drop` flag on `qri save`
+The current master branch has a `--drop` flag on the save command for "dropping components of a dataset". This RFC repurposes the `--drop` language to be about datasets you don't own, so this flag should be renamed to `--remove`. We shouldn't use `--delete`, because the word delete is now reserved for editing history.
```

Comment:
Does this still apply? I think we can keep using `save --drop [component]` now, and I think I'm more fond of that than changing the flag to `save --remove [component]`.

Reply:
agreed. removed